packages/react-menu/src/components/MenuTrigger/useTriggerElement.ts

```diff
@@ -81,41 +62,39 @@ export const useTriggerElement = (state: UseTriggerElementState): MenuTriggerSta
     if (openOnHover && !openOnContext) {
       setOpen(e, { open: true, keyboard: false });
     }
+
     child.props?.onMouseEnter?.(e);
   });
 
-  // no mouse leave, since mouse enter sets focus for menu items
-  const onBlur = useEventCallback((e: React.FocusEvent<HTMLElement>) => {
-    if (isOutsideMenu({ menuPopoverRef, triggerRef, event: e })) {
+  const onMouseLeave = useEventCallback((e: React.MouseEvent<HTMLElement>) => {
+    if (openOnHover && !openOnContext) {
       setOpen(e, { open: false, keyboard: false });
     }
 
-    child.props?.onBlur?.(e);
+    child.props?.onMouseLeave?.(e);
   });
 
   const disabled = child.props?.disabled;
   const triggerProps: MenuTriggerChildProps = {
     'aria-haspopup': true,
     'aria-expanded': open,
-    id: triggerId,
-    // spread props here because below event handlers must handle original prop
-    ...child.props,
+    id: child.props.id || triggerId,
 
     ...(!disabled
       ? {
           onClick,
           onMouseEnter,
+          onMouseLeave,
           onContextMenu,
           onKeyDown,
-          onBlur,
         }
       : // Spread disabled event handlers to implement contract and avoid specific disabled logic in handlers
         {
           onClick: noop,
           onMouseEnter: noop,
+          onMouseLeave: noop,
           onContextMenu: noop,
           onKeyDown: noop,
```

Comment:
Since `noop` is quite a common used term and function, what about adding it to `@fluentui/react-utilities`?